proposals/0029-cooperative-vector.md

```diff
@@ -289,32 +289,41 @@ guaranteed to be supported on all implementations can be found in
 [Minimum Support Set].
 
 
-### Reduce Sum Accumulate
+### Vector InterlockedAdd
 
 #### Syntax
 
 ``` llvm
-declare void @dx.op.vecreducesumacc.v[NUM][TY](
-    immarg i32,       ; opcode
-    <[NUM] x [TY]>,   ; input vector
-    %dx.types.Handle, ; output array resource 
-    i32)              ; output array offset
+; overloads: SM6.9: vector types
+; returns: void
+declare void @dx.op.atomicBinOp.v[NUM][TY](
+    i32,                  ; opcode
+    %dx.types.Handle,     ; resource handle
+    i32,                  ; binary operation code: ADD
+    i32,                  ; coordinate c0
+    i32,                  ; coordinate c1
+    i32,                  ; coordinate c2
+    <[NUM] x [TY]>)       ; input vector
 ```
 
 #### Overview
 
-Accumulates the components of a vector component-wise atomically (with device
-scope) to the corresponding elements of an array in memory. See note in
-[Atomic Operations].
+Adds new vector overloads to `@dx.op.atomicBinOp` to support accumulation of a
+vector component-wise atomically (with device scope) to the corresponding
+elements of an array in memory. See note in[Atomic Operations].
```

Comment:
I don't like that we're reusing atomicBinOp for an operation that doesn't have a full scope of atomicity. I'd rather we just scalarize to a bunch of separate atomicBinOp calls for each element rather than declare an overload that takes a vector and isn't atomic across the full input.

Reply:
That's a fair point. We would like implementations to be able to use any vector atomics that they can support though and choose the units of atomicity. Should we keep the InterlockedAdd override, but introduce a new dxil op?

Reply:
Another opcode would be an option, but we probably need to have a conversation about the name since it isn't atomic.

I suggest that we remove this part of the change from this PR so that we can get the rest in while we discuss what to do with ReduceSumAccumulate?